mlrun/execution.py

```diff
@@ -634,7 +634,8 @@ def log_artifact(
         :param viewer:        Kubeflow viewer type
         :param target_path:   Absolute target path (instead of using artifact_path + local_path)
         :param src_path:      Deprecated, use local_path
-        :param upload:        Upload to datastore (default is True)
+        :param upload:        Upload to datastore (default is True). If logging a directory, it will not be
+                              uploaded unless this flag is set to `True`.
```

Comment:
I see the default is set to `None` not `True`, is this intentional: `(default is True)`?

Reply:
> I see the default is set to `None` not `True`, is this intentional: `(default is True)`?

Yes, because in the implementation, we upload by default unless it is a directory artifact
https://github.com/mlrun/mlrun/blob/a192b22dcaa51d64ba1248f30b8bb1f8b3df8384/mlrun/artifacts/manager.py#L302C9-L302C10

Reply:
okay, I think this could be a bit confusing since the default value is `None`, but the behavior acts as if it's `True` when not specified. 
How about: 
`:param upload:        Upload to datastore. If not provided, and the target is not a directory, upload occurs by default. Directories are uploaded only when this flag is explicitly set to `True`.`
or something like that?